improve logging by inspecting exceptions

yaauie opened the pull request:
Notably, `Exception`s like `Faraday::SSLError` do not provide a helpful
`to_s` method, so when our logger flattens the context with `to_s`,
the exception with helpful message and cause is squashed down to
only include its class name.

guyboertje commented:
@yaauie 

I would like to merge #106 before this so it goes out with this one, OK?

yaauie commented:
Updated changelog.

I also changed version bump to be a minor, since this release will add the new feature from #106.

I'll merge/publish once the build completes.
